File: crates/auto-lang/test/a2c/095_storage_module/storage_module.c
#include "storage_module.at.h"


Storage_vtable Heap_Storage_vtable = {
    .data = Heap_Data
    .capacity = Heap_Capacity
    .try_grow = Heap_TryGrow
};

Storage_void__vtable Heap_Storage_void__vtable = {
    .data = Heap_Data
    .capacity = Heap_Capacity
    .try_grow = Heap_TryGrow
};


Storage_vtable InlineInt64_Storage_vtable = {
    .data = InlineInt64_Data
    .capacity = InlineInt64_Capacity
    .try_grow = InlineInt64_TryGrow
};

Storage_int_vtable InlineInt64_Storage_int_vtable = {
    .data = InlineInt64_Data
    .capacity = InlineInt64_Capacity
    .try_grow = InlineInt64_TryGrow
};



int main(void) {

    unknown heap_list = List<int, Heap>.new();
    unknown heap_cap = heap_list.capacity();


    unknown inline_list = List<int, InlineInt64>.new();
    unknown inline_cap = inline_list.capacity();

    
    return 0;
}

File: crates/auto-lang/test/a2r/114_storage_module/storage_module.rs
trait Storage {
    fn data(&self) -> *mut T;
    fn capacity(&self) -> /* unknown */;
    fn try_grow(&self, min_cap: /* unknown */) -> bool;
}


struct Heap {
    ptr: *mut T,
    cap: /* unknown */,
}

impl Heap {




}

impl Storage for Heap {
    fn data(&self) -> *mut () {
{
        }    }
    fn capacity(&self) -> /* unknown */ {
{
        }    }
    fn try_grow(&self, min_cap: /* unknown */) -> bool {
{
        }    }
}

struct InlineInt64 {
    buffer: [i32; 64],
}

impl InlineInt64 {




}

impl Storage for InlineInt64 {
    fn data(&self) -> *mut i32 {
{
        }    }
    fn capacity(&self) -> /* unknown */ {
{
        }    }
    fn try_grow(&self, min_cap: /* unknown */) -> bool {
{
        }    }
}

struct List {
    len: /* unknown */,
    store: S,
}

fn main() {

    let heap_list: List = List<int, Heap>.new();
    let heap_cap = heap_list.capacity();


    let inline_list: List = List<int, InlineInt64>.new();
    let inline_cap = inline_list.capacity();

    return 0;
}
